### ISSUE TITLE
redirectTo allows url which should not be permitted

### ISSUE THREAD

**Issue by awalias** [supabase/gotrue#107]
with 
```
      GOTRUE_SITE_URL: http://localhost:9999
      GOTRUE_URI_ALLOW_LIST: https://supabase.io/docs
```

and a `redirect_to` of `https://supabase.io/docss` is permitted, but it should instead redirect to the default `http://localhost:9999` (note the extra 's' in 'docss')

**Comment by awalias** [comment]
fixed by #111

**Comment by Illyism** [comment]
I think it would make sense to to disable
`https://supabase.io/docss` but to allow `https://supabase.io/docs?test=123` so that it's possible to add additional query strings to the redirect URI
